pp_netlib/network.py

```diff
@@ -175,91 +184,92 @@ def prune(self):
-        #                                             "\tScore\t\t\t\t\t" + "{:.4f}".format(scores[0]),
-        #                                             "\tScore (w/ betweenness)\t\t\t" + "{:.4f}".format(scores[1]),
-        #                                             "\tScore (w/ weighted-betweenness)\t\t" + "{:.4f}".format(scores[2])])
-        #                                             + "\n")
-        # sys.stderr.write(summary_contents)
-        # summary_file_name = os.path.join(self.outdir, (str(summary_file_prefix) + ".txt"))
-
-        # #################################
-        # #   extra little thing to       #
-        # #  write summary to plain text  #
-        # #################################
-
-        # with open(summary_file_name, "w") as summary:
-        #     summary.write(summary_contents)
-        # summary.close()
-        return
+            raise RuntimeError("Graph not found.")
+
+        summary_contents = summarise(self.graph, self.backend)
+
+        sys.stderr.write(summary_contents)
+
+         #################################
+        #  write summary to plain text  #
+        #################################
+        if summary_file_prefix is not None:
+            summary_file_name = os.path.join(self.outdir, (str(summary_file_prefix) + ".txt"))
+
+            with open(summary_file_name, "w") as summary:
+                summary.write(summary_contents)
+            summary.close()
 
     def visualize(self):
         # code for calling viz functions
         print("visualizing network")
         return #files associated with viz
 
     def load_network(self, network_file):
-        """Load the network based on input options
-       Returns the network as a graph-tool format graph, and sets
-       the slope parameter of the passed model object.
-
-       ## DEPENDS ON Fns: {none}
-
-       Args:
-            network_file (str)
-                Network file name
-            use_gpu (bool)
-                Use cugraph library to load graph
-       Returns:
-            genome_network (graph)
-                The loaded network
-    """
+        """Load a premade graph from a network file. 
+
+        Args:
+            network_file (str/path): The file in which the prebuilt graph is stored. Must be a .gt file, .graphml file or .xml file.
+        """
         # Load the network from the specified file
+        file_name, file_extension = os.path.splitext(network_file)
+        if file_extension in [".graphml", ".xml"]:
+            if self.backend == "GT":
+                loaded_graph = self.gt.load_graph(network_file)
+                num_nodes = len(list(loaded_graph.vertices()))
+                num_edges = len(list(loaded_graph.edges()))
+            if self.backend == "NX":
+                loaded_graph = self.nx.read_graphml(network_file)
+                num_nodes = len(loaded_graph.nodes())
+                num_edges = len(loaded_graph.edges())
+
+        if file_extension == ".gt":
```

Comment:
Put this if clause first, make the others `elif`, add and `else` clause if file extension unknown

Reply:
done